afew/tests/test_mailmover.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: ISC
+# Copyright (c) Johannes Löthberg <johannes@kyriasis.com>
+
+import datetime
+import email
+from email.utils import format_datetime, localtime, make_msgid
+import mailbox
+import notmuch
+import os
+import shutil
+import tempfile
+import unittest
+
+
+def get_notmuch_db(db_dir):
+    return notmuch.Database(db_dir, mode=notmuch.Database.MODE.READ_WRITE)
+
+def create_mail(msg, maildir, notmuch_db, tags, old=False):
+    if old:
+        date = format_datetime(localtime() - datetime.timedelta(days=20))
```

Comment:
For the sake of reproducibility, can we hardcode some random date in here, instead of using the current time?

Reply:
We could, but what would we then pass to `MailMover` as the `max_age` argument?  The only real option would be to then also add the functionality to tell MailMover a fixed date as the `max_age` value.

I'm not sure how this would affect reproducibility though?

Reply:
To diff multiple test run outputs.

Can we use `localtime()` in the test, but use `@freeze_time` from [freezegun](https://pypi.org/project/freezegun/)?

Reply:
Hm, interesting!  I'll try that out.

Reply:
@flokli There are some issues with this due to the fact that freezegun apparently doesn't handle timezones well.

With the following decorator:

```
@freeze_time("2019-01-30 12:00:00", auto_tick_seconds=15, tz_offset=0)
```

```
now = datetime.now()                 # 2019-01-30 12:00:00
now.strftime('%s')                   # 1548846000
```

And the problem here is that `1548846000` is 12:00 in UTC+1, which is 11:00 UTC.

Meanwhile the date in the mail ends up as `Date: Wed, 30 Jan 2019 13:01:45 +0100` which correctly is 12:00 UTC in UTC+1.  (If switching to `datetime.now()` instead of `localtime()` you get `Date: Wed, 30 Jan 2019 12:01:45 -0000`.)

This means that the date email is an hour ahead of the time returned by `datetime.now()`, which means that it's not included in the timestamp range passed to notmuch.

Reply:
I guess the solution would be to set a static date for the mails created, and it won't be a problem.

Reply:
@flokli 0b37e21 is my attempt at this now.  Is it sufficiently understandable?